src/main/java/teammates/ui/controller/PageData.java

```diff
@@ -322,10 +322,10 @@ private void ___________methods_to_generate_student_links() {
      * @return "Yet to Join" or "Joined"
      */
     public String getStudentStatus(StudentAttributes student) {
-        if (student.googleId == null || student.googleId.equals("")) {
-            return Const.STUDENT_COURSE_STATUS_YET_TO_JOIN;
-        } else {
+        if (student.isRegistered()) {
             return Const.STUDENT_COURSE_STATUS_JOINED;
+        } else {
+            return Const.STUDENT_COURSE_STATUS_YET_TO_JOIN;
```

Comment:
This looks like a duplicate of `getStudentStatus()` in `StudentAttributes`.
Is it necessary or can you redirect all of the calls to `getStudentStatus()` (of both classes) to either one of them?